### PR TITLE
[CoW Protocol] Implement partial fill into cow_protocol ethereum trades

### PR DESCRIPTION
**Pull request by gentrexha**
Brief comments on the purpose of your changes:

For the launch of partial fill orders, we're adding a partial_fill column to our trades label to differentiate between these types of trades.

cc @bh2smith for internal review

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### CODE REVIEW

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql [hunk 1]
```diff
@@ -220,7 +220,8 @@ valued_trades as (
            limit_buy_amount,
            valid_to,
            flags,
-           case when (flags % 2) = 0 then 'SELL' else 'BUY' end as order_type
+           case when (flags % 2) = 0 then 'SELL' else 'BUY' end as order_type,
+           case when ((flags / 2) % 2) = 0 then 'Not partially fillable' else 'Partially fillable' end as partial_fill
```
Comment (bh2smith) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:224]: I am not in favour of using a string here. Would you mind making this Boolean? the column name should suffice as in indication of what it means.

**Review by bh2smith — CHANGES_REQUESTED** [review]

File: tests/cow_protocol/ethereum/cow_protocol_ethereum_assert_partial_fill.sql [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+WITH unit_test1 AS
+(
+    SELECT
+        *
+    FROM
+        {{ ref('cow_protocol_ethereum_trades') }}
+    WHERE
+        order_uid = '0xaa1568c867c991bd462bcb6ee5939e4e01144d4e9e29bb74c2fa8d50b3afc92c519b70055af55a007110b4ff99b0ea33071c720a64289604'
+        and partial_fill != true
+),
```
Comment (olgafetisova) [inline tests/cow_protocol/ethereum/cow_protocol_ethereum_assert_partial_fill.sql:11]: Are these the test addresses from our team?

Comment (gentrexha) [reply tests/cow_protocol/ethereum/cow_protocol_ethereum_assert_partial_fill.sql:11]: Yes. They're order_uid's from Ben's [query](https://dune.com/queries/2321165).

**Review by bh2smith — APPROVED** [review]
Looks great!

**Review by olgafetisova — APPROVED** [review]

File: models/cow_protocol/ethereum/cow_protocol_ethereum_schema.yml [hunk 3]
```diff
@@ -136,6 +136,9 @@ models:
       - &order_type
         name: order_type
         description: "SELL/BUY string indicating if the trade was a sell or buy order"
+      - &partial_fill
+        name: partial_fill
+        description: "Boolean indicating if the order was partially filled"
```
Comment (Hosuke) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_schema.yml:141]: I add the new column in the schema, the description may not be correct.

**Review by Hosuke — APPROVED** [review]
LGTM.✅

File: models/cow_protocol/ethereum/cow_protocol_ethereum_schema.yml (hunk 3 above)
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_schema.yml:141]: So close, Partially "fillable"

Comment (gentrexha) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_schema.yml:141]: We've named it `partial_fill` in our SQL code.

Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_schema.yml:141]: perhaps this is more clear.

```suggestion
      - &partial_fill
        name: partial_fill
        description: "Boolean indicating if the order is partially fill-able"
```

Comment (gentrexha) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_schema.yml]: Indeed it is. Thanks Ben!

**Review by jeff-dude — APPROVED** [review]

**Comment by gentrexha** [comment]
Thanks @jeff-dude !

**Comment by gentrexha** [comment]
How long does it take for such a big table to update? If I try to select `partial_fill` from the `cow_Protocol_ethereum.trades` table it doesn't work for me. Query: https://dune.com/queries/2341402

cc @jeff-dude @Hosuke

**Comment by jeff-dude** [comment]
> How long does it take for such a big table to update? If I try to select `partial_fill` from the `cow_Protocol_ethereum.trades` table it doesn't work for me. Query: https://dune.com/queries/2341402
> 
> cc @jeff-dude @Hosuke

i need to push a change to our internal infra repo to have this column reflected in dunesql. if it helps, it's available in sparksql now until merged for dunesql.

**Comment by jeff-dude** [comment]
works now 👍

**Comment by gentrexha** [comment]
Thanks!

**Pull request merged**